apps/glusterfs/app_device.go

```diff
@@ -350,3 +350,92 @@ func (a *App) DeviceSetState(w http.ResponseWriter, r *http.Request) {
 		return "", nil
 	})
 }
+
+func (a *App) DeviceResync(w http.ResponseWriter, r *http.Request) {
+
+	vars := mux.Vars(r)
+	deviceId := vars["id"]
+
+	var (
+		device *DeviceEntry
+		node   *NodeEntry
+	)
+
+	// Get device info from DB
+	err := a.db.View(func(tx *bolt.Tx) error {
+		var err error
+		device, err = NewDeviceEntryFromId(tx, deviceId)
+		if err != nil {
+			return err
+		}
+		node, err = NewNodeEntryFromId(tx, device.NodeId)
+		if err != nil {
+			return err
+		}
+		return nil
+	})
+	if err == ErrNotFound {
+		http.Error(w, err.Error(), http.StatusNotFound)
+		return
+	} else if err != nil {
+		http.Error(w, err.Error(), http.StatusInternalServerError)
+		logger.Err(err)
+		return
+	}
+
+	logger.Info("Checking for device %v changes", deviceId)
+
+	// Check and update device in background
+	a.asyncManager.AsyncHttpRedirectFunc(w, r, func() (seeOtherUrl string, e error) {
+
+		// Get actual device info from manage host
+		info, err := a.executor.GetDeviceInfo(node.ManageHostName(), device.Info.Name, device.Info.Id)
+		if err != nil {
+			return "", err
+		}
+
+		if device.Info.Storage.Free == info.Size {
```

Comment:
The confusion comes from `getVgSizeFromNode` function (source for Info.Size) that returns **free** disk size and not total disk size.

For instance we started heketi on disk 10Gb.
In this case `getVgSizeFromNode` returns **10Gb** and Total = 10Gb, Free = **10Gb**, Used = 0Gb.
If we allocate 2Gb then Total = 10Gb, Free = **8Gb**, Used = 2Gb and `getVgSizeFromNode` returns **8Gb**.

So the first point to keep in mind is that there is only one real value that we can use as baseline is Storage.Free. Maybe write that line as: 
```
device.Info.Storage.Total == info.Size + device.Info.Storage.Used
```
Will it make more sense? I admit that total size is more natural to comprehend and I was struggling with that line a lot.